### PR TITLE
rbd-mirror: check remote image mirroring state when bootstrapping

### PR DESCRIPTION
**Pull request by trociny**
Fixes: http://tracker.ceph.com/issues/18447
Signed-off-by: Mykola Golub <mgolub@mirantis.com>

### CODE REVIEW

**Comment by trociny** [comment]
@dillaman If you agree with the fix I will remade it with async calls (for both mirror_image_get and is_tag_owner).

**Comment by dillaman** [comment]
@trociny lgtm

**Comment by trociny** [comment]
@dillaman converted to async

File: src/tools/rbd_mirror/image_replayer/IsPrimaryRequest.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+   * <start>
+   *    |
+   *    v
+   * GET_MIRROR_STATE * * * * *
+   *    |                     *
+   *    v                     *
+   * IS_TAG_OWNER * * * * * * * (error)
+   *    |                     *
+   *    v                     *
+   * <finish> * * * * * * * * *
```
Comment (dillaman) [inline src/tools/rbd_mirror/image_replayer/IsPrimaryRequest.h]: Nit: arrow "<" pointer to finish state

**Comment by trociny** [comment]
@dillaman updated

**Comment by trociny** [comment]
@dillaman I just noticed there is still is_tag_owner sync call. I think it would be a good idea to update it to use IsPrimaryRequest too?

**Comment by dillaman** [comment]
@trociny You mean in ```OpenLocalImageRequest```? Sounds good to me to replace it while working here.

**Comment by trociny** [comment]
@dillaman Yes, sorry, I meant `OpenLocalImageRequest`. Updated. 

Not sure only if it is safe that now we check primary status no holding owner_lock?

**Comment by dillaman** [comment]
@trociny The lock should have only been there for protecting access to ```ImageCtx::exclusive_lock```

**Pull request merged**